### PR TITLE
Allow for Quantities with null values. Catch invalid values

### PR DESCRIPTION
**Pull request by hossenlopp**
In the Quantity constructor, allow for value to be null. Also throw errors for invalid values. ie. prevent NaN values

Pull requests into cql-execution require the following.
Submitter and reviewer should ✔ when done.
For items that are not-applicable, mark "N/A" and ✔.

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository. 
It is strongly recommended to include a person from each of those projects as a reviewer.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] Code coverage has not gone down and all code touched or added is covered.
- [ ] All dependent libraries are appropriately updated or have a corresponding PR related to this change

**Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: test/elm/quantity/test.coffee [hunk 1]
```diff
@@ -33,3 +33,12 @@ describe 'Quantity', ->
   it 'should allow creation of Quantity with empty string unit', ->
     should.doesNotThrow ->
       new Quantity({unit: "", value: 9})
+
+  it 'should allow for the value of the quantity to be null', ->
```
Comment (losborne) [inline test/elm/quantity/test.coffee]: What do you think about adding another test for `undefined` like this (mostly as a way to document that this function behaves this way and to catch any changes that might work for `null` but not `undefined` in the future:

```
  it 'should convert undefined values to null', ->
    should.doesNotThrow ->
      q = new Quantity({unit: "mg", value: undefined})
      should.equal(q.value, null)
```
